### ISSUE TITLE
Capitalisation of package names in Pipfile is inconsistent.

### ISSUE THREAD

**Issue by funkybob** [pypa/pipenv#1855]
When I run "pipenv install {package}" all the package names in the Pipfile are lower-cased.

However, when I run "pipenv uninstall {package}" many of the names are changed back to their mixed case form.

This creates a *lot* of noise in change logs.

$ python -m pipenv.help
<details><summary>$ python -m pipenv.help output</summary>

Pipenv version: `'11.9.0'`

Pipenv location: `'/home/curtis/.local/lib/python2.7/site-packages/pipenv'`

Python location: `'/usr/bin/python'`

Other Python installations in `PATH`:

  - `2.7`: `/usr/bin/python2.7`
  - `2.7`: `/usr/bin/python2.7`
  - `3.4`: `/usr/bin/python3.4m`
  - `3.4`: `/usr/bin/python3.4`

  - `2.7.9`: `/usr/bin/python`
  - `2.7.9`: `/usr/bin/python2`
  - `3.4.2`: `/usr/bin/python3`

PEP 508 Information:

```
{'implementation_name': 'cpython',
 'implementation_version': '0',
 'os_name': 'posix',
 'platform_machine': 'x86_64',
 'platform_python_implementation': 'CPython',
 'platform_release': '3.16.0-5-amd64',
 'platform_system': 'Linux',
 'platform_version': '#1 SMP Debian 3.16.51-3+deb8u1 (2018-01-08)',
 'python_full_version': '2.7.9',
 'python_version': '2.7',
 'sys_platform': 'linux2'}
```

System environment variables:

  - `PYTHONDONTWRITEBYTECODE`
  - `PGPASSWORD`
  - `LOGNAME`
  - `USER`
  - `HOME`
  - `PATH`
  - `PYTHONUNBUFFERED`
  - `LANG`
  - `TERM`
  - `SHELL`
  - `LANGUAGE`
  - `SHLVL`
  - `PGUSER`
  - `PYTHONPATH`
  - `PIP_PYTHON_PATH`
  - `_`
  - `SSH_CONNECTION`
  - `SSH_TTY`
  - `OLDPWD`
  - `SSH_CLIENT`
  - `PWD`
  - `MAIL`
  - `LS_COLORS`

Pipenv–specific environment variables:

Debug–specific environment variables:

  - `PATH`: `/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games`
  - `SHELL`: `/bin/bash`
  - `LANG`: `en_AU.UTF-8`
  - `PWD`: `/home/curtis/review_app/workspace/candle-server/src/components/erasmus`

---------------------------

</details>

------------

##### Expected result

Consistent casing for both install / uninstall.

##### Actual result

```diff
 flower = "*"
-flask = "*"
-flask-xml-rpc = "*"
-flask-restful = "*"
+Flask = "*"
+Flask-XML-RPC = "*"
+Flask-RESTful = "*"
 pytest-flask = "*"
```

**Comment by techalchemy** [comment]
Seems to be occurring since #1314 -- as @ncoghlan suggested there we should just never alter what is in the Pipfile, but normalize it on the way in

**Comment by techalchemy** [comment]
This is fixed in #1826 and released, sorry for the bug!
